### PR TITLE
Fixes Borgs and Syndicats getting hurt by glass shards and other things + honkbot slipping

### LINKED ISSUE #27492: Borgs are taking damage walking on Glass Shards
Description

#26764 Seems to have made them take damage but not slip. Compared to not doing either before
**Reproduction**

1. Be entity with `StepTriggerImmune`
2. Walk on glass
3. Observe your damage

### PR DESCRIPTION
**Pull request by Vermidia**
## About the PR

Borgs and Syndicats no longer get hurt by things that require shoes

## Why / Balance

🐛
Fixes #27492 and #27664

## Technical details

Removed the StepTriggerImmune event - that would require the glass to have StepTriggerImmune (which makes no sense!) and StepTriggerImmuneComponent itself (Anything that was given it was expected to be immune to glass shards, but not like, mines)
Instead ClothingRequiredStepTrigger now checks if the tripper has ClothingRequiredStepTriggerImmune component on itself or if its in an inventory slot: If it has either they're safe.

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase
deliciosu glass

## Breaking changes

Remove StepTriggerImmuneComponent entirely - anything that uses it will need to use the ClothingRequiredStepTriggerImmune component instead.

**Changelog**

:cl:
- fix: Borgs and other creatures that shouldn't get hurt stepping on things no longer get hurt stepping on things.
- fix: Honkbots slip again

### CODE REVIEW

**Comment by FungiFellow** [comment]
`ClothingRequiredStepTriggerImmune`
`ClothingStepTriggerImmune`

These do both sound pretty similar.

File: Content.Shared/StepTrigger/Systems/StepTriggerImmuneSystem.cs [hunk 1]
```diff
@@ -12,19 +11,13 @@ public sealed class StepTriggerImmuneSystem : EntitySystem
...
     {
-        SubscribeLocalEvent<StepTriggerImmuneComponent, StepTriggerAttemptEvent>(OnStepTriggerAttempt);
-        SubscribeLocalEvent<ClothingRequiredStepTriggerComponent, StepTriggerAttemptEvent>(OnStepTriggerClothingAttempt);
+        SubscribeLocalEvent<ClothingRequiredStepTriggerComponent, StepTriggerAttemptEvent>(OnStepTriggerAttempt);
         SubscribeLocalEvent<ClothingRequiredStepTriggerComponent, ExaminedEvent>(OnExamined);
     }
 
-    private void OnStepTriggerAttempt(Entity<StepTriggerImmuneComponent> ent, ref StepTriggerAttemptEvent args)
+    private void OnStepTriggerAttempt(EntityUid uid, ClothingRequiredStepTriggerComponent component, ref StepTriggerAttemptEvent args)
     {
```
Comment (metalgearsloth) [inline Content.Shared/StepTrigger/Systems/StepTriggerImmuneSystem.cs:19]: Why are you reverting code conventions changes.

Comment (Vermidia) [reply Content.Shared/StepTrigger/Systems/StepTriggerImmuneSystem.cs]: My bad, I just renamed OnStepTriggerClothingAttempt and it had it set up this way.

**Comment by Vermidia** [comment]
Turns out this fixes honkbot slipping too

**Comment by UbaserB** [comment]
played honkbot yesterday and was in despair when i couldnt slip (i wasnt able to assist my syndicate master by slipping sec)

**Comment by Plykiya** [comment]
> `ClothingRequiredStepTriggerImmune` `ClothingStepTriggerImmune`
> 
> These do both sound pretty similar.
> 
> Other than that, this should solve the issue well enough.

aren't these the exact same thing...?

**Comment by Vermidia** [comment]
No because full step trigger immunity = they can't trigger mines. That's why I renamed it.

**Comment by Vermidia** [comment]
Ok, I deleted the old StepTriggerImmune component entirely and all the behavior is just through checking ClothingRequiredStepTriggerImmuneComponent.

**Comment by Plykiya** [comment]
> No because full step trigger immunity implies they can't trigger mines (also puddles are handled by something else, this clothing step trigger immunity doesn't stop you from slipping), which isn't anything given the component is trying to do (not trigger glass shards or dice on the floor). That's why I renamed it. The later _does_ work but does mean if the entity is worn it'll also just give whatever clothing step trigger immunity (which might be weird if its on your head or something)
> 
> Edit: Actually the later may be doable if I check the whole clothing slots thing.

no, puddles definitely do use StepTriggers and the SlipperySystem

**Comment by Vermidia** [comment]
Borgs don't because they have the NoSlipComponent. Syndicats do slip.

**Review by AJCM-git — CHANGES_REQUESTED** [review]
ClothingRequiredStepTriggerImmune is now very intuitive because things like borgs and syndicats have em but both of them have (almost) no clothing, either rename the component or something else

**Comment by Vermidia** [comment]
Changed ClothingRequiredStepTrigger PreventableStepTriggerComponent and ClothingRequiredStepTriggerImmune to ProtectedFromStepTriggersComponent, which I hope gets the point across generally without making them seem immune to all step triggers.

**Comment by Plykiya** [comment]
god I love naming conventions

**Comment by Cojoke-dot** [comment]
mmm... Landmines will not work when stepped on by Borgs, its kinda niche but also kinda of annoying

**Comment by Vermidia** [comment]
No, they still will
This isn't full complete step trigger immunity, just to some step triggers. I tried to change the name to reflect that

**Comment by Vermidia** [comment]
I cooked the branch, check out #31011

**Pull request closed without merging**